app/pseudocone.py

```diff
@@ -67,6 +67,41 @@ def ListTestDataUsers(self, request, context):
         pseudocone_response = convert_json_list_to_pseudocone_response(filtered_resource_type)
         return pseudocone_response
 
+    def ListTestDataUsersBetweenDates(self, request, context):
+
+        if not request.end_interaction_time:
```

Comment:
these tests are good! we could do some refactoring and add them in separate functions that will be used by all endpoints to avoid repeating code (this applies for all code blocks in the endpoint). It could also be in future work so no worries for now.